scripts/migration/cjs2esm

```diff
@@ -0,0 +1,154 @@
+#!/usr/bin/env node
+
+const fs = require('fs');
+const path = require('path');
+
+const filenames = process.argv.slice(2); // Trim off node and script name.
+
+/** Absolute path of repository root. */
+const repoPath = path.resolve(__dirname, '..', '..');
+
+//////////////////////////////////////////////////////////////////////
+// Process files mentioned on the command line.
+//////////////////////////////////////////////////////////////////////
+
+/** RegExp matching require statements. */
+const requireRE =
+  /(?:const\s+(?:([$\w]+)|(\{[^}]*\}))\s+=\s+)?require\('([^']+)'\);/g;
+
+/** RegExp matching key: value pairs in destructuring assignments. */
+const keyValueRE = /([$\w]+)\s*:\s*([$\w]+)\s*(?=,|})/g;
+
+/** Prefix for RegExp matching a top-level declaration. */
+const declPrefix =
+    '(?:const|let|var|(?:async\\s+)?function(?:\\s+\\*)?|class)';
+
+for (const filename of filenames) {
+  let contents = null;
+  try {
+    contents = String(fs.readFileSync(filename));
+  } catch (e) {
+    console.error(`error while reading ${filename}: ${e.message}`);
+    continue;
+  }
+  console.log(`Converting ${filename} from CJS to ESM...`);
+
+  // Remove "use strict".
+  contents = contents.replace(/^\s*["']use strict["']\s*; *\n/m, '');
+
+  // Migrate from require to import.
+  contents = contents.replace(
+    requireRE,
+    function (
+      orig, // Whole statement to be replaced.
+      name, // Name of named import of whole module (if applicable).
+      names, // {}-enclosed list of destructured imports.
+      moduleName, // Imported module name or path.
+    ) {
+      if (moduleName[0] === '.') {
+        // Relative path.  Could check and add '.mjs' suffix if desired.
+      }
```

Comment:
Yes.  There is a bit of optional code that I might write to save having to fix up import paths manually—the only difficulty is that sometimes the imported filename will be an `.mjs` and other times it will just be `.js`, and not even testing file existence can necessarily distinguish because the file in question might be mid-conversion by the same invocation of this tool.  So I left this as a placeholder for the moment.